src/artifactRegistry.ts

```diff
@@ -224,27 +240,22 @@ export type DockerTag = {
  *
  * @param {string} prevTag
  *   The previous docker tag: of the following format: main---0013567-2024.04-g<githash>
+ *
  * @param {string} nextTag
  *  The next docker tag: main---0013567-2024.04-g<githash>
  *
  * @param {DockerTag[]} dockerTags
  *  The docker tags as we need to parse and filter into relevant commits. Provided by a previous call to the artifact registry.
  *
- *
- * @param {function} getTagFromDockerTag
- *   Should map from `projects/platform-cross-environment/locations/us-central1/repositories/platform-docker/packages/servicename/tags/2022.02-278-g123456789` -> `2022.02-278-g123456789`
- *   Should be a wrapper around the ArtifactRegistryClient, but written this way for testability, so the behavior can be injected.
- *
  * @returns {string[]} - The relevant commits as strings
  */
 export function getRelevantCommits(
   prevTag: string,
   nextTag: string,
   dockerTags: DockerTag[],
-  getTagFromDockerTag: (dockerTag: DockerTag) => string,
 ): string[] {
-  if (!prevTag.startsWith('main')) return [];
-  if (!nextTag.startsWith('main')) return [];
+  if (!isMainTag(prevTag)) return [];
```

Comment:
I do think we should differentiate between "no changes" and "the promotion is from or to a tag that isn't of the form `main---*` and so we can't tell you what changes are being published". Maybe return null in this case?  (Could also go into the next PR where we bubble up the results.)

Reply:
I'll merge this for now.